src/Nethermind/Nethermind.Mev/Source/BundlePool.cs

```diff
@@ -18,34 +18,61 @@
 using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Drawing;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
 using Nethermind.Blockchain;
+using Nethermind.Blockchain.Find;
 using Nethermind.Core;
 using Nethermind.Core.Collections;
 using Nethermind.Core.Crypto;
 using Nethermind.Core.Extensions;
 using Nethermind.Int256;
+using Nethermind.JsonRpc;
+using Nethermind.Logging;
 using Nethermind.Mev.Data;
 using Nethermind.Mev.Execution;
+using Nethermind.TxPool.Collections;
+using Org.BouncyCastle.Security;
+using ILogger = Nethermind.Logging.ILogger;
 
 namespace Nethermind.Mev.Source
 {
+
     public class BundlePool : IBundlePool, ISimulatedBundleSource, IDisposable
     {
         private readonly IBlockFinalizationManager? _finalizationManager;
+        private readonly ITimestamper _timestamper;
+        private readonly IMevConfig _mevConfig;
         private readonly IBlockTree _blockTree;
         private readonly IBundleSimulator _simulator;
-        private readonly SortedRealList<MevBundle, ConcurrentBag<Keccak>> _bundles = new(MevBundleComparer.Default);
+        private readonly SortedPool<MevBundle, MevBundle, long> _bundles;
```

Comment:
maybe a simple conurrentdictionary -> int -> bundle?

Reply:
Would have to be ConcurrentDictionary<int, ConcurrentBag<MevBundle>> and... its definitely worse for eviction on Capacity